.github/workflows/ci.yml

```diff
@@ -89,7 +89,9 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Clang version
-        run: clang --version
+        run: |
+          xcode-select --switch "/Applications/Xcode15.1.app"
```

Comment:
```suggestion
          sudo xcode-select --switch "/Applications/Xcode15.1.app"
```

Reply:
If any version is going to be picked, shouldn't it be the one we use for releases?

Reply:
Or `sudo xcode-select -switch /Applications/Xcode_15.1.app`. It works in https://github.com/hebasto/bitcoin/actions/runs/7385516810/job/20090426555

Reply:
>  Run sudo xcode-select --switch "/Applications/Xcode15.1.app"
xcode-select: error: invalid developer directory '/Applications/Xcode15.1.app'

Reply:
> If any version is going to be picked, shouldn't it be the one we use for releases?

Xcode 15.0.1 (15A507) and 15.1 (15C65)  versions are available now, and they are subjects for frequent updates.

Reply:
> Xcode 15.0.1 (15A507) and 15.1 (15C65) versions are available now, 

My suggestion was to pick `15.0`, which we use for release builds, so that CI better matches our release process, rather than `15.1`, as you suggested. Otherwise code could compile in the CI, but not in Guix.

Reply:
> Looks like the macOS CI now fails for unrelated reasons. If someone created a separate pull request to bump xcode in the macOS CI, that'd be great.

The underlying issue is that Apple Clang 15 does not support `-Xclang -internal-isystem` anymore. Therefore, supporting for the `--enable-suppress-external-warnings` option on `x86_64` macOS becomes a non-trivial task.

Reply:
Right, the `event2/event.h` is no longer considered as a system header, which allows a compiler to emit warnings from it, even configured with the `--enable-suppress-external-warnings` option .

Reply:
We could pass `-nostdinc` to the compiler and specify include ditrectories explicitly. However, such a solution looks a bit ugly to me.

Reply:
> The underlying issue is that Apple Clang 15 does not support -Xclang -internal-isystem anymore.

I'd be very suprised if this was the case, given it's a core Clang flag? Even more suprised if they'd for some reason only removed it for x86_64?

Reply:
> > The underlying issue is that Apple Clang 15 does not support -Xclang -internal-isystem anymore.
> 
> I'd be very suprised if this was the case, given it's a core Clang flag? Even more suprised if they'd for some reason only removed it for x86_64?

https://github.com/bitcoin/bitcoin/actions/runs/7395607754/job/20119092833:
```
checking whether C++ preprocessor accepts -Xclang -internal-isystem/usr/local/include... no
```

Reply:
Pretty sure it's still there? i.e: https://github.com/apple/llvm-project/blob/94624bdf65b8ad61f9bf3948a5bf387c6fa0a967/clang/lib/Driver/ToolChain.cpp#L1138 ?

Reply:
Here are excerpts from `config.log`:
- Apple clang version 14.0.3 (clang-1403.0.22.14.1):
```
configure:23855: checking whether C++ preprocessor accepts -Xclang -internal-isystem/usr/local/include
configure:23875: g++ -std=c++20 -E  -Werror -Xclang -internal-isystem/usr/local/include conftest.cpp
configure:23875: $? = 0
configure:23885: result: yes
```
- Apple clang version 15.0.0 (clang-1500.0.40.1):
```
configure:23855: checking whether C++ preprocessor accepts -Xclang -internal-isystem/usr/local/include
configure:23875: g++ -std=c++20 -E  -Werror -Xclang -internal-isystem/usr/local/include conftest.cpp
error: unknown argument: '-internal-isystem/usr/local/include'
```

Reply:
> Again, this should affect everyone on macOS today already on current master, using clang-15 (or later), so a separate issue and bugfix can be filed?

See https://github.com/bitcoin/bitcoin/issues/29174.